backend/queries/volunteers.js

```diff
@@ -83,6 +91,29 @@ const getAllVolunteers = async (vEmail, company, skill, name, publicProfilesOnly
       }
     }
 
+    for (let volunteer of volunteersList) {
```

Comment:
a more efficient way is have the select query do this instead of doing it in a loop with `JSON_AGG`
we don't have to worry about deleting falsy values since they are handled in the frontend, in your code you have `if (interest.mentoring)  ...`